packages/functions/src/service.ts

```diff
@@ -247,12 +263,16 @@ async function call(
   // Default timeout to 70s, but let the options override it.
   const timeout = options.timeout || 70000;
 
+  const failAfterHandle = failAfter(timeout);
   const response = await Promise.race([
     postJSON(url, body, headers, functionsInstance.fetchImpl),
-    failAfter(timeout),
+    failAfterHandle.promise,
     functionsInstance.cancelAllRequests
   ]);
 
+  // Always clear the failAfter timeout
```

Comment:
Oh I was going to suggest putting this in the _delete method but I see that that _delete will resolve cancelAllRequests and bring us here, so that's fine.

Reply:
Yup